app/scripts/modules/core/src/managed/ArtifactDetail.tsx

```diff
@@ -23,6 +24,139 @@ function shouldDisplayResource(name: string, type: string, resource: IManagedRes
   return !!resource.moniker && name === resource.artifact?.name && type === resource.artifact?.type;
 }
 
+interface IEnvironmentProps {
+  application: Application;
+  environment: {
+    name: string;
+    state: 'current' | 'deploying' | 'approved' | 'pending' | 'previous' | 'vetoed';
+    deployedAt?: string;
+    replacedAt?: string;
+    replacedBy?: string;
+    statefulConstraints: IStatefulConstraint[];
+  };
+  resourcesByEnvironment: { [environment: string]: IManagedResourceSummary[] };
+  type: string;
+  version: string;
+}
+const Environment = memo(
```

Comment:
I'm going to work on moving this whole `useEnvironmentTypeFromResources` usage into `EnvironmentRow` (which I personally feel is the right place for this computation) in a separate PR as I don't want to further delay the MVP launch.

Reply:
I take this back. It wasn't a big change as I anticipated, so I've updated this PR.